lib/nerves_hub_web/channels/device_channel.ex

```diff
@@ -345,4 +328,32 @@ defmodule NervesHubWeb.DeviceChannel do
   end
 
   defp should_audit_log?(_join_reply, _params), do: true
+
+  defp device_matches_deployment_payload?(device, payload) do
+    payload.active &&
+      device.product_id == payload.product_id &&
+      device.firmware_metadata.platform == payload.platform &&
+      device.firmware_metadata.architecture == payload.architecture &&
+      Enum.all?(payload.conditions["tags"], &Enum.member?(device.tags, &1)) &&
+      Deployments.version_match?(device, payload)
+  end
+
+  defp assign_deployment(socket, device, payload) do
+    device =
+      device
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_change(:deployment_id, payload.id)
```

Comment:
```suggestion
      |> Ecto.Changeset.put_change(:deployment_id, payload.id)
```
Pretty sure you can skip this and go straight to adding a change to the struct?